### PR TITLE
Fix edit message error via front-end "lock"

### PR DESCRIPTION
**Pull request by lasryaric**

### CODE REVIEW

File: front/components/assistant/conversation/AgentSuggestion.tsx [hunk 1]
```diff
@@ -22,6 +22,7 @@ export function AgentSuggestion({
   userMessage: UserMessageType;
   conversation: ConversationType;
 }) {
+  const editingMessage = useRef(0);
```
Comment (spolu) [inline front/components/assistant/conversation/AgentSuggestion.tsx:25]: Should we do `isEditingMessage` state which is more aligned with what we do eslewhere?

Comment (lasryaric) [reply front/components/assistant/conversation/AgentSuggestion.tsx:25]: `isEditingMessage` naming is definitely better. But by state you meant a ref (from `useRef`) right?

Comment (spolu) [reply front/components/assistant/conversation/AgentSuggestion.tsx:25]: I was more thinking like using a `useState` and prevent the click on the button which is what we do elsewhere. Less direct but equally effective no?

File: front/components/assistant/conversation/AgentSuggestion.tsx [hunk 2]
```diff
@@ -74,29 +75,40 @@ export function AgentSuggestion({
...
+            }),
+          }
+        );
 
-    if (!mRes.ok) {
-      const data = await mRes.json();
-      window.alert(`Error adding mention to message: ${data.error.message}`);
+        if (!mRes.ok) {
+          const data = await mRes.json();
+          window.alert(
```
Comment (spolu) [inline front/components/assistant/conversation/AgentSuggestion.tsx:103]: Can you change that into a notifications while we're at it :pray:

File: front/components/assistant/conversation/AgentSuggestion.tsx [hunk 3]
```diff
@@ -74,29 +75,40 @@ export function AgentSuggestion({
...
+            `Error adding mention to message: ${data.error.message}`
+          );
+        }
+      } finally {
+        isEditingMessage.current--;
+      }
+    } else {
+      console.error("already editing message");
```
Comment (spolu) [inline front/components/assistant/conversation/AgentSuggestion.tsx:111]: Let's remove the console?

**Review by spolu — CHANGES_REQUESTED** [review]

File: front/components/assistant/conversation/AgentSuggestion.tsx (hunk 1 above)
Comment (lasryaric) [reply front/components/assistant/conversation/AgentSuggestion.tsx]: as discussed on Slack, we are moving forward with useRef().

**Comment by lasryaric** [comment]
@spolu PTAL?

**Review by spolu — APPROVED** [review]
:100:

**Pull request merged**